proxy/src/config.rs

```diff
@@ -340,6 +339,42 @@ fn parse_number<T>(s: &str) -> Result<T, ParseError> where T: FromStr {
     s.parse().map_err(|_| ParseError::NotANumber)
 }
 
+fn parse_duration(s: &str) -> Result<Duration, ParseError> {
+    use regex::Regex;
+    lazy_static! {
+        static ref RE: Regex = Regex::new(r"(?x)^
+            \s* (\d+) \s*
+            ( ms | msecs? | millis?  | milliseconds?
```

Comment:
I think we should only support the short form of each duration unit that we support: "ms", "s", "m", "h", "d", to make life easier and to make sure our testing is comprehensive.